nav2_rviz_plugins/include/nav2_rviz_plugins/nav2_panel.hpp

```diff
@@ -113,6 +132,14 @@ private Q_SLOTS:
   rclcpp::Subscription<nav2_msgs::action::NavigateThroughPoses::Impl::GoalStatusMessage>::SharedPtr
     nav_through_poses_goal_status_sub_;
 
+  // map_pose subscriber for initial pose
+  rclcpp::Subscription<geometry_msgs::msg::PoseWithCovarianceStamped>::SharedPtr
+    map_pose_sub_;
+
+  // global_costmap subscriber
+  nav2_costmap_2d::CostmapSubscriber * costmap_sub_;
+  std::shared_ptr<nav2_costmap_2d::Costmap2D> costmap_;
+
```

Comment:
The `collision_checker_'` uses the costmap and the footprint polygon to determine if a pose is valid. 

Your current implementation does this but does not check the footprint. 

To reduce code, and prevent creating another instance of the costmap you can just use the `collision_checker_` by `costmap_sub_ = std::make_unique<nav2_costmap_2d::CostmapSubscriber>(
    shared_from_this(), costmap_topic);
  footprint_sub_ = std::make_unique<nav2_costmap_2d::FootprintSubscriber>(
    shared_from_this(), footprint_topic, *tf_, robot_base_frame, transform_tolerance);

  collision_checker_ = std::make_shared<nav2_costmap_2d::CostmapTopicCollisionChecker>(
    *costmap_sub_, *footprint_sub_, this->get_name())`
    
   I should have been a bit more careful with my line placement :).

Reply:
Gottcha!

Reply:
The only concern in adding the collision_checker_ for me is that, we then need to maintain configs for the rav2_rviz_plugins. This is mainly because some of the parameters in the collision_checker_ might be robot specific such as the robot_base_frame, footprint topic and transform_tolerance. This would actually add an extra configuration step for the user while setting up the robot. 

Maybe that is the way to go forward, if we want to make RViz a bit more user-friendly.

Reply:
We might also want to add the collision checker to the waypoint follower, for the people who use wp_follower standalone. Would be good for a new issue right @jwallace42 ?

Reply:
Yeah that sounds good :).